scripts/install.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 Rapyuta Robotics
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Function to print usage
+### Prints the usage instructions and exits
+usage() {
+  echo -e "\033[1;31mUsage: $0 [<release tag>]\033[0m"
+  echo -e "\033[1;31mIf no release tag is specified, the latest release will be used.\033[0m"
+  echo -e "\033[1;31mExample: $0 v1.0.0\033[0m"
+  exit 1
+}
+
+# Check if more than one argument is provided
+### Ensure only one argument is provided, otherwise print usage and exit
+if [ "$#" -gt 1 ]; then
+  usage
+fi
+
+# Set the repository base URL (hardcoded)
+### Base URL for the GitHub repository releases API
+BASE_URL="https://api.github.com/repos/rapyuta-robotics/rapyuta-io-cli/releases"
+
+# Get the tag or assume the latest release if no tag is provided
+### Handle release tag argument
+if [ "$#" -eq 0 ]; then
+  echo -e "\033[1;34m⏳ Downloading the latest release...\033[0m"
+  API_URL="$BASE_URL/latest"
+else
+  echo -e "\033[1;34m⏳ Downloading release $1...\033[0m"
+  TAG="$1"
+  API_URL="$BASE_URL/tags/$TAG"
+fi
+
+# Fetch release information
+### Fetch release data from GitHub API
+RELEASE_DATA=$(curl -s "$API_URL")
+
+# Extract the asset URL
+### Extract the browser download URL for the AppImage asset
+ASSET_URL=$(echo "$RELEASE_DATA" | grep '"browser_download_url":' | grep '.AppImage' | sed -E 's/.*"browser_download_url":\s*"(https:[^"]*)".*/\1/')
+
+# Check if the asset URL was found
+### Ensure the asset URL was extracted successfully
+if [ "$ASSET_URL" = "null" ] || [ -z "$ASSET_URL" ]; then
+  echo -e "\033[1;31mError: No assets found in release \"$TAG\" for the repository rapyuta-robotics/rapyuta-io-cli\033[0m"
+  exit 1
+fi
+
+# Set the temporary download location in /tmp
+### Temporary download location
+TEMP_PATH="/tmp/rio.download"
+
+# Check if the asset already exists in /tmp
+### Remove existing asset if it exists
+if [ -f "$TEMP_PATH" ]; then
+  sudo rm -f "$TEMP_PATH"
+fi
+
+# Download the asset with a progress bar
+### Download the AppImage asset
+curl -fSL -o "$TEMP_PATH" --progress-bar "$ASSET_URL"
+
+# Make the asset executable
+### Set execute permissions on the downloaded asset
+chmod +x "$TEMP_PATH"
+
+# Move the asset to /usr/local/bin
+### Move the asset to the system bin directory
+sudo mv "$TEMP_PATH" /usr/local/bin/rio
```

Comment:
suggestion: Consider using the `install(1)` command.

```bash
sudo install -C "$TMP_PATH" /usr/local/bin/rio
```

Reply:
Done.